src/backend/distributed/commands/table.c

```diff
@@ -3222,6 +3171,17 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 					{
 						if (columnConstraint->contype == CONSTR_IDENTITY)
 						{
+							/*
+							 * We currently don't support adding an identity column for an MX table
+							 */
```

Comment:
Why don't we allow this? Is it because `worker_modify_identity_columns` touches all the identity columns? If so, maybe it should take the column names too.

Reply:
That's not quite the case. Similar to what's been done in here: https://github.com/citusdata/citus/blob/main/src/backend/distributed/commands/table.c#L3175-L3176 I want to implement parallel behavior for identity columns because of similar reasons.